src/stactools/core/utils/raster_footprint.py

```diff
@@ -0,0 +1,313 @@
+            in degrees.
+        preserve_topology (bool): Preserve topology during simplification.
+        no_data(Optional[int]): explicitly set the no data value if not in image metadata. If
+            set to None, this will return the footprint including no data values.
+
+    Returns:
+        List[Tuple[float, float]]: a list of the densified points
+    """
+    src = rasterio.open(href)
+
+    # create datamask
+    if no_data is None:
+        no_data = src.nodata
+
+    scaled_out_shape = (int(src.shape[0] / scale), int(src.shape[1] / scale))
+    arr = src.read(1, out_shape=scaled_out_shape)
+    data_val = 1 if no_data != 1 else 0
+
+    if no_data is not None:
+        arr[numpy.where(arr != no_data)] = data_val
+    else:
+        arr.fill(data_val)
+
+    shapes_transform = src.transform * Affine.scale(scale)
+
+    data_polygons = [
+        shape(polygon_dict)
+        for polygon_dict, region_value in rasterio.features.shapes(
+            arr, transform=shapes_transform
+        )
+        if region_value == data_val
+    ]
+
+    if not data_polygons:
+        return None
+    elif len(data_polygons) == 1:
+        polygon = data_polygons[0]
+    else:
+        polygon = MultiPolygon(data_polygons).convex_hull
+
+    polygon = densify_reproject_simplify(
+        polygon,
+        src.crs,
+        densification_factor=densification_factor,
+        precision=precision,
+        simplify_tolerance=simplify_tolerance,
+        preserve_topology=preserve_topology,
+    )
+
+    import json
+
+    print(json.dumps(mapping(polygon)))
+
+    return mapping(polygon)  # type: ignore
+
+
+def densify_reproject_simplify(
+    polygon: Polygon,
+    crs: CRS,
+    *,
+    densification_factor: Optional[int] = None,
+    precision: int = DEFAULT_PRECISION,
+    simplify_tolerance: Optional[float] = None,
+    preserve_topology: bool = True,
+) -> Polygon:
+    """
+        From the input Polygon, densifies the polygon, reprojects it to EPSG:4326, and then
+        simplifies the resulting polygon.
+
+        See :py:meth:`update_geometry_from_asset_footprint` for more details about densification
+        and simplification.
+
+     Args:
+        polygon (Polygon): The input Polygon
+        crs (CRS): The CRS of the input Polygon
+        densification_factor (Optional[int]): The factor by which to increase point density
+            within the polygon. A factor of 2 would double the density of points (placing one
+            new point between each existing pair of points), a factor of 3 would place two points
+            between each point, etc.
+        precision (int): The number of decimal places to include in the coordinates for the
+            reprojected geometry. Defaults to 3 decimal places.
+        simplify_tolerance (Optional[float]): The minimum distance to maintain between points,
+            in degrees.
+        preserve_topology (bool): Preserve topology during simplification.
```

Comment:
I believe preserving topology is only relevant for MultiPolygons (prevent overlaps and gaps between Polygons in a MultiPolygon when simplifying). As it stands, MultiPolygons are not used (a single convex hull Polygon is being simplified), so this option has no effect.